### PR TITLE
Add instructions how to configure Bitbucket Server oAuth2

### PR DESCRIPTION
**Pull request by vinokurig**
## What does this pull request change?
Add instructions how to configure Bitbucket Server oAuth2 for the factory flow.
depends on https://github.com/eclipse-che/che-server/pull/440
## What issues does this pull request fix or reference?
https://github.com/eclipse/che/issues/21949

## Specify the version of the product this pull request applies to
next
## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/main/src/constants.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+.Prerequisites
+
+* You are logged in to the Bitbucket Server.
+* link:https://www.gnu.org/software/coreutils/base64[`base64`] is installed in the operating system you are using.
+
+.Procedure
+
+. Go to *Administration > Applications > Application links*.
+. Select *Create link*.
+. Select *External application*, and then choose *Incoming* as the direction.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]: Nitpick:
```suggestion
. Select *External application* and *Incoming*.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+. Enter `pass:c,a,q[{prod-url}]/api/oauth/callback` to the *Redirect URL* field.
+. Chose `Admin - Write`  checkbox in the `Application permissions` section.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Select the *Admin - Write*  checkbox in *Application permissions*.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+. Click the *Save* button.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Click *Save*.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<github_oauth_client_id>__' | base64
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
$ echo -n '__<Bitbucket_Client_ID>__' | base64
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+----
+
+. Copy the *Client Secret* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<github_oauth_client_secret>__' | base64
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
$ echo -n '__<Bitbucket_Client_secret>__' | base64
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 6]
```diff
@@ -0,0 +1,58 @@
+:_content-type: PROCEDURE
+:description: Applying an oauth-2 application link Secret for the Bitbucket Server
+:keywords: bitbucket, bitbucket-server, application-link, bitbucket-secret
+:navtitle: Applying an application link Secret for the Bitbucket Server
+// :page-aliases:
+
+[id="applying-an-oauth-2-application-link-secret-for-the-bitbucket-server"]
+= Applying an oauth-2 application link Secret for the Bitbucket Server
+
+Prepare and apply an application link Secret for the Bitbucket Server.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]: Nitpick: we already mentioned this Secret on this page.
```suggestion
Prepare and apply the application link Secret for the Bitbucket Server.
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+.Prerequisites
+
+* The application link is set up in the Bitbucket Server.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]: Nitpick:
```suggestion
* The application link is set up on the Bitbucket Server.
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+* The Base64-encoded values, which were generated when setting up the Bitbucket application link, are prepared:
+** Bitbucket Client ID
+** Bitbucket Client Secret
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]: Nitpick: 
```suggestion
** Bitbucket *Client ID*
** Bitbucket *Client secret*
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    che.eclipse.org/scm-server-endpoint: __<bitbucket_server_url>__ <2>
+type: Opaque
+data:
+  id: __<Base64_Bitbucket_Client_ID>__ <3>
+  secret: __<Base64_Bitbucket_Client_Secret>__ <4>
+----
+<1> The {prod-short} namespace. The default is `{prod-namespace}`.
+<2> The URL of the Bitbucket Server.
+<3> The Base64-encoded *Bitbucket Client ID*.
+<4> The Base64-encoded *Bitbucket Client Secret*.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]:
```suggestion
<3> The Base64-encoded Bitbucket *Client ID*.
<4> The Base64-encoded Bitbucket *Client secret*.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+. Chose `Admin - Write`  checkbox in the `Application permissions` section.
+. Click the *Save* button.
+. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<github_oauth_client_id>__' | base64
+----
+
+. Copy the *Client Secret* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link App Secret:
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+.Procedure
+
+. Go to *Administration > Applications > Application links*.
+. Select *Create link*.
+. Select *External application*, and then choose *Incoming* as the direction.
+. Enter `pass:c,a,q[{prod-url}]/api/oauth/callback` to the *Redirect URL* field.
+. Chose `Admin - Write`  checkbox in the `Application permissions` section.
+. Click the *Save* button.
+. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket Server OAuth App Secret:
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+. Select the *Admin - Write*  checkbox in *Application permissions*.
+. Click *Save*.
+. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket application link Secret:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<Bitbucket_Client_ID>__' | base64
+----
+
+. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link App Secret:
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 13]
```diff
@@ -0,0 +1,58 @@
+:_content-type: PROCEDURE
+:description: Applying an oauth-2 application link Secret for the Bitbucket Server
+:keywords: bitbucket, bitbucket-server, application-link, bitbucket-secret
+:navtitle: Applying an application link Secret for the Bitbucket Server
+// :page-aliases:
+
+[id="applying-an-oauth-2-application-link-secret-for-the-bitbucket-server"]
+= Applying an oauth-2 application link Secret for the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]:
```suggestion
= Applying an OAuth 2.0 application link Secret for the Bitbucket Server
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 14]
```diff
@@ -0,0 +1,58 @@
+:_content-type: PROCEDURE
+:description: Applying an oauth-2 application link Secret for the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]:
```suggestion
:description: Applying an OAuth 2.0 application link Secret for the Bitbucket Server
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+:keywords: bitbucket, bitbucket-server, application-link, bitbucket-secret
+:navtitle: Applying an application link Secret for the Bitbucket Server
+// :page-aliases:
+
+[id="applying-an-oauth-2-application-link-secret-for-the-bitbucket-server"]
+= Applying an oauth-2 application link Secret for the Bitbucket Server
+
+Prepare and apply the application link Secret for the Bitbucket Server.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]:
```suggestion
Prepare and apply the OAuth 2.0 application link Secret for the Bitbucket Server.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 16]
```diff
@@ -0,0 +1,41 @@
+:_content-type: PROCEDURE
+:description: Setting up an oauth-2 application link on the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
:description: Setting up an OAuth 2.0 application link on the Bitbucket Server
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+:keywords: bitbucket, bitbucket-server, application-link, oauth2
+:navtitle: Setting up an oauth-2 application link on the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
:navtitle: Setting up an OAuth 2.0 application link on the Bitbucket Server
```

File: modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc [hunk 18]
```diff
@@ -0,0 +1,58 @@
+:_content-type: PROCEDURE
+:description: Applying an oauth-2 application link Secret for the Bitbucket Server
+:keywords: bitbucket, bitbucket-server, application-link, bitbucket-secret
+:navtitle: Applying an application link Secret for the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_applying-an-oauth-2-application-link-secret-for-the-bitbucket-server.adoc]:
```suggestion
:navtitle: Applying an OAuth 2.0 application link Secret for the Bitbucket Server
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+// :page-aliases:
+
+[id="setting-up-an-oauth-2-application-link-on-the-bitbucket-server"]
+= Setting up an oauth-2 application link on the Bitbucket Server
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
= Setting up an OAuth 2.0 application link on the Bitbucket Server
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+
+Set up an application link for OAuth 2.0 on the Bitbucket Server.
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
Set up an OAuth 2.0 application link on the Bitbucket Server.
```

File: modules/administration-guide/pages/configuring-oauth-2-for-a-bitbucket-server.adoc [hunk 21]
```diff
@@ -0,0 +1,17 @@
+:_content-type: ASSEMBLY
+:description: Configuring OAuth 2.0 for a Bitbucket Server
+:keywords: configuring-oauth, authorization, bitbucket, bitbucket-server, oauth-2, oauth-2.0
+:navtitle: Configuring OAuth 2.0 for a Bitbucket Server
+// :page-aliases:
+
+[id="configuring-oauth-2-for-a-bitbucket-server"]
+= Configuring OAuth 2.0 for a Bitbucket Server
+
+To enable users to work with a remote Git repository that is hosted on a Bitbucket Server:
```
Comment (max-cx) [inline modules/administration-guide/pages/configuring-oauth-2-for-a-bitbucket-server.adoc]:
```suggestion
You can use OAuth 2.0 to enable users to work with a remote Git repository that is hosted on a Bitbucket Server:
```

File: modules/administration-guide/pages/configuring-oauth-2-for-a-bitbucket-server.adoc [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+. Set up an application link (OAuth 2.0) on the Bitbucket Server.
```
Comment (max-cx) [inline modules/administration-guide/pages/configuring-oauth-2-for-a-bitbucket-server.adoc]:
```suggestion
. Set up an OAuth 2.0 application link on the Bitbucket Server.
```

File: modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+. Select the *Admin - Write*  checkbox in *Application permissions*.
+. Click *Save*.
+. Copy the *Client ID* and encode it to Base64 for use when applying the Bitbucket application link Secret:
++
+[subs="+quotes,+attributes,+macros"]
+----
+$ echo -n '__<Bitbucket_Client_ID>__' | base64
+----
+
+. Copy the Bitbucket *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```
Comment (max-cx) [inline modules/administration-guide/partials/proc_setting-up-an-oauth-2-application-link-on-the-bitbucket-server.adoc]:
```suggestion
. Copy the *Client secret* and encode it to Base64 for use when applying the Bitbucket application link Secret:
```

**Review by ibuziuk — APPROVED** [review]

**Comment by vinokurig** [comment]
@max-cx any updates? Is it OK to go?

**Review by l0rd — APPROVED** [review]

**Pull request merged**